### PR TITLE
[Test] Delete Single Reply 

### PR DESCRIPTION
**Pull request by The-indigo**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**

Fixes #issue_number (replace issue_number with the issue number this PR solves)

Summarise the main tasks handled in the pull request
Using jest to test for the delete a reply endpoint
**description of Task to be completed?**

Describe (in detail) what the task completed in the pull request does as per the relevant task

**How should this be manually tested?**

Give bullet point instructions on how to setup the code and test it manually (Assume no prior experience on JS or code)

**Any background context you want to provide?**

Any pertinent information that should be considered

**What is the link to the issue on Github?**

[Issue #362](https://github.com/microapidev/comment-microapi/issues/362)

**Questions:**

If something is unclear or you want some questions to be addressed by your peers, mention them here

### CODE REVIEW

**Comment by simply-alliv** [comment]
Your build is failing.

File: __test__/api/replies/deleteSinglereply.test.js [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    const res = await request
+      .delete(url)
+      .set("Authorization", bearerToken)
+      .send({
+        ownerId: expectedR.ownerId,
+      });
+    expect(res.status).toBe(200);
+    expect(res.body.status).toEqual("success");
+    expect(res.body.data).toEqual({
```
Comment (piouson) [inline __test__/api/replies/deleteSinglereply.test.js:65]: add `matchers` to check db that comment no longer has deleted replies, something like..

```
res = DELETE /reply/url
comment = await model.findbyid(commentId)
expect(comment.replies).notContains(replyId)
```

File: __test__/api/replies/deleteSinglereply.test.js [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  // 422 validation error
+  test("Should return validation error when ownerId is missing", async () => {
+    const url = `/v1/comments/${comment.commentId}/replies/${reply.replyId}`;
+    const bearerToken = `bearer ${global.appToken}`;
+
+    const res = await request
+      .delete(url)
+      .set("Authorization", bearerToken)
+      .send({
+        ownerId: reply.ownerId,
```
Comment (piouson) [inline __test__/api/replies/deleteSinglereply.test.js:98]: `ownerId` should be missing for 422 error..?

File: __test__/api/replies/deleteSinglereply.test.js [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  beforeEach(async () => {
+    comment = new CommentModel({
+      content: "A comment from user 1",
+      ownerId: "user1@email.com",
+      origin: "123123",
+      refId: 2,
+      applicationId: global.application._id,
+    });
+
+     reply= new ReplyModel({
```
Comment (piouson) [inline __test__/api/replies/deleteSinglereply.test.js:22]: fix lint errors

File: __test__/api/replies/deleteSinglereply.test.js [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+      ownerId: "user2@email.com",
+      commentId: comment.id,
+    });
+
+    //save dummy comment
+    await comment.save();
+    await reply.save();
+
+    // Add replies to the dummy comment
+    await CommentModel.findByIdAndUpdate(comment.id, {replies: reply.id});
```
Comment (piouson) [inline __test__/api/replies/deleteSinglereply.test.js:33]: fix lint errors `npm run lint:fix`

**Comment by The-indigo** [comment]
Restarting this issue on another pr

File: __test__/api/replies/deleteSinglereply.test.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+      ownerId: "user2@email.com",
+      commentId: dummyComment.id,
+    });
+
+    //save dummy comment,reply
+    const savedC = await dummyComment.save();
+    const savedR = await dummyReply.save();
+
+    // Add replies to the dummy comment
+    await CommentModel.findByIdAndUpdate(savedC.id, { replies: savedR.id });
```
Comment (piouson) [inline __test__/api/replies/deleteSinglereply.test.js:34]: this does not correctly add `replyId` to `comment.replies`
review your work or use 
```
const savedR = await dummyReply.save();
dummyComment.replies.push(savedR.id);
const savedC = await dummyComment.save();
```

File: __test__/api/replies/deleteSinglereply.test.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+      .send({
+        ownerId: reply.ownerId,
+      });
+    expect(res.status).toEqual(200);
+    expect(res.body.status).toEqual("success");
+    expect(res.body.data).toMatchObject(reply);
+
+    //add matchers to check db that comment no longer has deleted replies
+    // const comms = await CommentModel.findById(comment.commentId);
+    //  expect(comms.replies).notContains(reply.replyId)
```
Comment (piouson) [inline __test__/api/replies/deleteSinglereply.test.js:71]: compare results of cached comments before `DELETE` to result of comment after `DELETE`
```
await CommentModel.findById(commentId).then(item => {
  expect(commentHandler(item).numOfReplies).toHaveLength(comment.numOfReplies-1)
  // also check actual content does not contain replyId
  // refer to the Jest doc
});
```
[Jest doc](https://jestjs.io/docs/en/expect)

File: controllers/repliesController/deleteSingleReply.js [hunk 7]
```diff
@@ -51,17 +52,18 @@ const deleteSingleReply = async (req, res, next) => {
     if (!reply) {
       return next(new CustomError(404, "reply not found"));
     }
-    await Comments.findByIdAndUpdate(commentId, {
-      // it doesn't matter if the parent exist or not
-      $pull: { replies: replyId },
-    });
+    // await Comments.findByIdAndUpdate(commentId, {
```
Comment (piouson) [inline controllers/repliesController/deleteSingleReply.js:55]: please do not comment out the `findByIdAndUpdate` transactions

File: controllers/repliesController/deleteSingleReply.js [hunk 8]
```diff
@@ -51,17 +52,18 @@ const deleteSingleReply = async (req, res, next) => {
...
+    //   // it doesn't matter if the parent exist or not
+    //   $pull: { replies: replyId },
```
Comment (piouson) [inline controllers/repliesController/deleteSingleReply.js:57]: please do not comment out, I think this updates the `comment.replies` array in the db

**Comment by dave-ok** [comment]
Nice recovery!

**Pull request merged**